File: test/data/cpp/test03/StateIdle.cpp
#include "StateIdle.h"
StateIdle::StateIdle(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateIdle::~StateIdle()
{
}
BaseState* StateIdle::TransitEvConfig()
{
    _controllee->DoAction1();
    _stateController->InstanceOfConfiguring->Setup(false, false);
    return _stateController->InstanceOfConfiguring;
}
const char* StateIdle::GetStateName()
{
    return "NotShooting.Idle";
}
BaseState* StateIdle::GetParent()
{
    return _stateController->InstanceOfNotShooting;
}
int StateIdle::GetStateID()
{
    return 2;
}

File: test/data/cs/test03/StateIdle.cs
public class StateIdle : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateIdle(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitEvConfig()
    {
        _controllee.DoAction1();
        _stateController.InstanceOfConfiguring.Setup(false, false);
        return _stateController.InstanceOfConfiguring;
    }
    public override string GetStateName()
    {
        return "NotShooting.Idle";
    }
    public override BaseState? GetParent()
    {
        return _stateController.InstanceOfNotShooting;
    }
    public override int GetStateID()
    {
        return 2;
    }
}